### DISCUSSION TITLE
postgis function call in rpc - error

### DISCUSSION THREAD

**Discussion by rdgit** [supabase/supabase#2474] [Questions] [answered]
# can a postgis function called via .rpc work?

## using a function to return json, a 404 is returned when called via the rpc api.
the function call works in the supabase sql editor, when called with "SELECT"

the error is 42883: "function st_geometryfromtext(text) does not exist"
"hint": "No function matches the given name and argument types. You might need to add explicit type casts."

could it be a permission issue?

**Comment by steve-chavez** [comment, marked as answer]
> can a postgis function called via .rpc work?

Yes, it can. But since extensions are not part of the `public`(your api) schema, you have to create a wrapper function to expose some of them:

```sql
create or replace function "textToGeom"(txt text) returns geometry as $$
  select extensions.st_geometryfromtext(txt);
$$ language sql;
```

```js
const { data, error } = await supabase
  .rpc('textToGeom', { txt: "POINT(-71.064544 42.28787)" })
```

**Reply by rdgit** [reply]
Thanks Steve, this is great intel; very much appreciate the example code as well.

The other thread I was following suggested to install the PostGIS extension in its own schema, then grant access to the api user as well.  What is your take on that approach?

Cheers - Cals

**Reply by steve-chavez** [reply]
> The other thread I was following suggested to install the PostGIS extension in its own schema, then grant access to the api user as well. What is your take on that approach?

Must have missed that one, can you share the link?

Personally, I wouldn't do it. There could be potentially expensive(slow) functions in PostGIS(IIRC, there are like 300 hundred functions there) and exposing that to clients doesn't seem safe. Ideally your api should whitelist the functions you want exposed.

**Reply by rdgit** [reply]
Thanks for this Steve - it's taken me a while to circle back to it!
